### PR TITLE
error handling, solution and testing for task 10-GET /api/users

### PR DESCRIPTION
**Pull request by DeeSzubert**
error handling, solution and testing for task 10-GET /api/users

### CODE REVIEW

File: controllers/users-controllers.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const { fetchAllUsers } = require("../models/users-models");
+
+function getAllUsers(request, response, next) {
+  fetchAllUsers().then((users) => {
+    response.status(200).send(users);
```
Comment (pippaamy) [inline controllers/users-controllers.js:5]: remember to send this users back on an object of users

File: models/users-models.js [hunk 2]
```diff
@@ -11,4 +11,10 @@ function checkIfUserExists(username) {
     });
 }
 
-module.exports = { checkIfUserExists };
+function fetchAllUsers() {
+  return db.query(`SELECT * FROM users`).then(({ rows }) => {
```
Comment (pippaamy) [inline models/users-models.js:15]: perfect!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -333,6 +333,23 @@ describe("/api/comments/:comment_id", () => {
   });
 });
 
+describe("/api/users", () => {
+  test("GET 200: Responds with an array of user objects, each of which should have 3 properties: username, name and avatar_url. ", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.length).toBe(4);
```
Comment (pippaamy) [inline __tests__/app.test.js:342]: this test is great!

**Review by pippaamy — COMMENTED** [review]
Great job, remember to send back the result on an object :)

**Comment by DeeSzubert** [comment]
Thank you Pippa

⁣Get BlueMail for Android ​

**Pull request merged**